From f5f4f507bd9b5cc9cb5b470c270b83b4d7d24b0f Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:12:50 -0400
Subject: [ethernet] i40e/i40evf: Force a shifted '1' to be unsigned

Message-id: <1409663591-32531-128-git-send-email-sassmann@redhat.com>
Patchwork-id: 90604
O-Subject: [RHEL7.1 PATCH 127/148] i40e/i40evf: Force a shifted '1' to be unsigned
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Paul M Stillwell Jr <paul.m.stillwell.jr@intel.com>

Force a shifted '1' to be unsiged to avoid shifting a signed int

Change-ID: I688cbd082af0f2e1df548fda25847a5ca04babcf
Signed-off-by: Paul M Stillwell Jr <paul.m.stillwell.jr@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 30fe8ad3667c9060a572e48ca4ada8f3161e5d2e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40e_hmc.h b/drivers/net/ethernet/intel/i40evf/i40e_hmc.h
index a2ad9a4..931c880 100644
--- a/drivers/net/ethernet/intel/i40evf/i40e_hmc.h
+++ b/drivers/net/ethernet/intel/i40evf/i40e_hmc.h
@@ -127,7 +127,7 @@ struct i40e_hmc_info {
   ((((type) == I40E_SD_TYPE_PAGED) ? 0 : 1) <<  \
   I40E_PFHMC_SDDATALOW_PMSDTYPE_SHIFT) |   \
   (1 << I40E_PFHMC_SDDATALOW_PMSDVALID_SHIFT);  \
- val3 = (sd_index) | (1 << I40E_PFHMC_SDCMD_PMSDWR_SHIFT); \
+ val3 = (sd_index) | (1u << I40E_PFHMC_SDCMD_PMSDWR_SHIFT); \
  wr32((hw), I40E_PFHMC_SDDATAHIGH, val1);   \
  wr32((hw), I40E_PFHMC_SDDATALOW, val2);    \
  wr32((hw), I40E_PFHMC_SDCMD, val3);    \
@@ -146,7 +146,7 @@ struct i40e_hmc_info {
   I40E_PFHMC_SDDATALOW_PMSDBPCOUNT_SHIFT) |  \
   ((((type) == I40E_SD_TYPE_PAGED) ? 0 : 1) <<  \
   I40E_PFHMC_SDDATALOW_PMSDTYPE_SHIFT);   \
- val3 = (sd_index) | (1 << I40E_PFHMC_SDCMD_PMSDWR_SHIFT); \
+ val3 = (sd_index) | (1u << I40E_PFHMC_SDCMD_PMSDWR_SHIFT); \
  wr32((hw), I40E_PFHMC_SDDATAHIGH, 0);    \
  wr32((hw), I40E_PFHMC_SDDATALOW, val2);    \
  wr32((hw), I40E_PFHMC_SDCMD, val3);    \
-- 
1.7.1